From 9f7453358e53665b4637d9dda98f948150628713 Mon Sep 17 00:00:00 2001
From: Zhixiong Chi <zhixiong.chi@windriver.com>
Date: Wed, 27 Mar 2019 01:40:21 +0000
Subject: [PATCH] proc: fix the uninitialized compiling warning

Move the assignment out of the condition area in the context, so
that we can avoid the follow compileing warning:
  task_mmu.c:763:7: warning: 'last_vma' may be used uninitialized

Signed-off-by: Zhixiong Chi <zhixiong.chi@windriver.com>
---
 fs/proc/task_mmu.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/fs/proc/task_mmu.c b/fs/proc/task_mmu.c
index 669787f..af47449 100644
--- a/fs/proc/task_mmu.c
+++ b/fs/proc/task_mmu.c
@@ -769,13 +769,14 @@ static int show_smap(struct seq_file *m, void *v, int is_pid)
 			mss->first_vma_start = vma->vm_start;
 			mss->first = false;
 		}
-		last_vma = !m_next_vma(priv, vma);
 	} else {
 		rollup_mode = false;
 		memset(&mss_stack, 0, sizeof(mss_stack));
 		mss = &mss_stack;
 	}
 
+	last_vma = !m_next_vma(priv, vma);
+
 	smaps_walk.private = mss;
 
 #ifdef CONFIG_SHMEM
-- 
1.7.9.5

